src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -1905,22 +1910,170 @@ protected virtual void TagHelperPrefixDirective()
                         errors.Add(duplicateDiagnostic);
                     }
 
-                    return new TagHelperPrefixDirectiveChunkGenerator(prefix, errors);
+                    var parsedDirective = ParseDirective(prefix, Span.Start, TagHelperDirectiveType.TagHelperPrefix, errors);
+
+                    return new TagHelperPrefixDirectiveChunkGenerator(
+                        prefix,
+                        parsedDirective.DirectiveText,
+                        parsedDirective.DirectiveStart,
+                        errors);
                 });
         }
 
+        // Internal for testing.
+        internal void ValidateTagHelperPrefix(
+            string prefix,
+            SourceLocation directiveLocation,
+            List<RazorDiagnostic> diagnostics)
+        {
+            foreach (var character in prefix)
+            {
+                // Prefixes are correlated with tag names, tag names cannot have whitespace.
+                if (char.IsWhiteSpace(character) || InvalidNonWhitespaceNameCharacters.Contains(character))
+                {
+                    diagnostics.Add(
+                        RazorDiagnostic.Create(
+                            new RazorError(
+                                Resources.FormatInvalidTagHelperPrefixValue(SyntaxConstants.CSharp.TagHelperPrefixKeyword, character, prefix),
+                                directiveLocation,
+                                prefix.Length)));
+
+                    return;
+                }
+            }
+        }
+
+        private ParsedDirective ParseDirective(
+            string directiveText,
+            SourceLocation directiveLocation,
+            TagHelperDirectiveType directiveType,
+            List<RazorDiagnostic> errors)
+        {
+            var offset = 0;
+            directiveText = directiveText.Trim();
+            if (directiveText.Length >= 2 &&
+                directiveText.StartsWith("\"", StringComparison.Ordinal) &&
+                directiveText.EndsWith("\"", StringComparison.Ordinal))
+            {
+                directiveText = directiveText.Substring(1, directiveText.Length - 2);
+                if (string.IsNullOrEmpty(directiveText))
+                {
+                    offset = 1;
```

Comment:
This is for the edge case like `@tagHelperPrefix ""`. When we strip off the quotes around this, we are left with `string.Empty` but the location should point to after the starting quote.